### PR TITLE
Fix CI (Backport #221 into 1.x)

### PR DESCRIPTION
**Pull request by martinRenou**
* Remove packageManager entry

* Fix for Python 3.12

* Remove use of pytest-asyncio

* Fix mypy

* Re-enable Python 3.11

* Don't test win/py3.12

* Fix URLs in package.json

* Update yarn.lock

* Revert previous commit

* Remove pip --pre

### CODE REVIEW

**Comment by martinRenou** [comment]
Ah `Fix License Headers` is running from the 1.x branch so it should pass once merged, not before

**Comment by davidbrochart** [comment]
Should https://github.com/jupyterlab/jupyter-collaboration/pull/224/commits/99874ebd9f9bb6fbf3b2e53b8fa65952c3eced4e be replaced with #199?

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -145,12 +145,7 @@ jobs:
         uses: jupyterlab/maintainer-tools/.github/actions/base-setup@v1
       - name: Install the Python dependencies
         run: |
-<<<<<<< HEAD
           pip install --pre -e ".[test]"
```
Comment (davidbrochart) [inline .github/workflows/test.yml]: I think you want the other one:
```py
pip install -e ".[test]"
```

**Review by davidbrochart — APPROVED** [review]
Thanks!

**Pull request merged**